File: app/src/main/java/ru/magflayer/spectrum/presentation/pages/main/MainView.java
package ru.magflayer.spectrum.presentation.pages.main;

import com.arellomobile.mvp.viewstate.strategy.OneExecutionStateStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import ru.magflayer.spectrum.presentation.common.mvp.view.BaseView;

interface MainView extends BaseView {

    @StateStrategyType(OneExecutionStateStrategy.class)
    void showToolbar(boolean showToolbar);

    @StateStrategyType(OneExecutionStateStrategy.class)
    void showFloatingButton(boolean showFloatingButton);

}

File: app/src/main/java/ru/magflayer/spectrum/presentation/pages/main/MainView.kt
package ru.magflayer.spectrum.presentation.pages.main

import moxy.viewstate.strategy.OneExecutionStateStrategy
import moxy.viewstate.strategy.StateStrategyType
import ru.magflayer.spectrum.presentation.common.mvp.view.BaseView

interface MainView : BaseView {

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun showToolbar(showToolbar: Boolean)

    @StateStrategyType(OneExecutionStateStrategy::class)
    fun showFloatingButton(showFloatingButton: Boolean)

}
